solver/monomial_solve.h

```diff
@@ -23,15 +23,18 @@
 #include "solver/matrix_mult_typedef.h"
 #include "solver/solver_params.h"
 #include "su3.h"
-
-int solve_degenerate(spinor * const P, spinor * const Q, solver_params_t solver_params,
-                     const int max_iter, double eps_sq, const int rel_prec, 
-                     const int N, matrix_mult f, int solver_type);
-
-int solve_mshift_oneflavour(spinor ** const P, spinor * const Q, solver_params_t* solver_params);
-
+#include"solver/matrix_mult_typedef.h"
+#include"solver/solver_params.h"
+#include"su3.h"
+int solve_degenerate(spinor * const P, spinor * const Q, solver_params_t solver_params, const int max_iter, 
```

Comment:
@kostrzewa I was also wondering: shall we make the call to solve_degenerate standard and turn it into `int solve_tm(spinor * const P, spinor * const Q, solver_params_t * solver_params);`
Now the structure solver_params is copied and the others variables are already included in that.